www/src/pages/editor/NewMineralSiteModal.tsx

```diff
@@ -0,0 +1,454 @@
+        );
+
+        setCountryOptions(
+          Array.from(countryStore.records.values())
+            .filter((country): country is Country => country !== null)
+            .map((country: Country) => ({ value: country.id, label: country.name }))
+        );
+
+        setStateOptions(
+          Array.from(stateOrProvinceStore.records.values())
+            .filter((state): state is StateOrProvince => state !== null)
+            .map((state: StateOrProvince) => ({ value: state.id, label: state.name }))
+        );
+
+        setDepositTypeOptions(
+          Array.from(depositTypeStore.records.values())
+            .filter((type): type is DepositType => type !== null)
+            .map((type) => ({ value: type.id ?? "", label: type.name ?? "Unnamed" }))
+        );
+
+        setUnitOptions(
+          Array.from(unitStore.records.values())
+            .filter((unit): unit is Unit => unit !== null)
+            .map((unit) => ({ value: unit.id ?? "", label: unit.name ?? "Unnamed" }))
+        );
+        if (visible) {
+          const generatedId = `record-${uuidv4()}`;
+          setRecordId(generatedId);
+          form.setFieldsValue({ recordId: generatedId });
+        }
+      } catch (error) {
+        console.error("Error fetching data for NewMineralSiteModal:", error);
+      }
+    };
+
+    fetchData();
+  }, [commodityStore, countryStore, stateOrProvinceStore, depositTypeStore, unitStore, form, visible]);
+
+
+  const handleSourceTypeChange = (e: RadioChangeEvent) => {
+    const value = e.target.value;
+    setSelectedSourceType(value);
+
+    const refDocUrl = form.getFieldValue("refDoc");
+    const currentUser = userStore.getCurrentUser()?.url;
+
+    if (value === "unpublished") {
+      form.setFieldsValue({ sourceId: `unpublished::${currentUser || "unknown"}` });
+    } else if (refDocUrl) {
+      form.setFieldsValue({ sourceId: `${value}::${refDocUrl}` });
+    } else {
+      form.setFieldsValue({ sourceId: null });
+    }
+  };
+
+  interface FormValues {
+    latitude?: number;
+    longitude?: number;
+    country?: string;
+    stateorprovince?: string;
+    name: string;
+    refDoc: string;
+    refComment?: string;
+    sourceType: string;
+    deposittype: string;
+    deposittypeconfidence: number;
+    tonnage?: number;
+    grade?: number;
+    gradeUnit?: string;
+    tonnageUnit?: string;
+  }
+
+
+  const handleSave = async (values: FormValues) => {
+    try {
+      let location = undefined;
+      if (values.latitude !== undefined && values.longitude !== undefined) {
+        location = `POINT (${values.longitude} ${values.latitude})`;
+      }
+
+      const countries = values.country
+        ? [
+          new CandidateEntity({
+            observedName: countryStore.getByURI(values.country)?.name,
+            source: values.country,
+            normalizedURI: countryStore.getByURI(values.country)?.uri,
```

Comment:
why call getByURI twice? also values.country is already URI? why need to make useless call in 125?

Reply:
need to fix similar codes